File: app/turnstile.py
from .env import getenv
import aiohttp

async def verify_captcha(*, token: str, ipaddr: str) -> bool:
    data = {
        "secret": getenv("turnstile"),
        "response": token,
        "ip": ipaddr
    }

    async with aiohttp.ClientSession() as session:
        async with session.post("https://challenges.cloudflare.com/turnstile/v0/siteverify", data=data) as response:
            return (await response.json()).get("success", False)

File: app/turnstile.ts
import axios from 'axios';
import { getenv } from './env';

export async function verifyCaptcha(token: string): Promise<boolean> {
    const data = {
        secret: getenv('turnstile_secret'),
        response: token
    };

    try {
        const response = await axios.post('https://challenges.cloudflare.com/turnstile/v0/siteverify', data);
        return response.data.success;
    } catch (error) {
        console.error('Error verifying captcha:', error);
        return false;
    }
}
